src/wp-includes/block-template-utils.php

```diff
@@ -1444,36 +1477,46 @@ function get_template_hierarchy( $slug, $is_custom = false, $template_prefix = '
  * @since 6.5.0
  * @access private
  *
- * @param stdClass        $post    An object representing a template or template part
+ * @param stdClass        $changes An object representing a template or template part
  *                                 prepared for inserting or updating the database.
  * @param WP_REST_Request $request Request object.
  * @return stdClass The updated object representing a template or template part.
  */
-function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
-	$filter_name = current_filter();
-	if ( ! str_starts_with( $filter_name, 'rest_pre_insert_' ) ) {
-		return $post;
-	}
-	$post_type = str_replace( 'rest_pre_insert_', '', $filter_name );
-
+function inject_ignored_hooked_blocks_metadata_attributes( $changes, $request ) {
 	$hooked_blocks = get_hooked_blocks();
 	if ( empty( $hooked_blocks ) && ! has_filter( 'hooked_block_types' ) ) {
-		return $post;
+		return $changes;
 	}
 
-	// At this point, the post has already been created.
-	// We need to build the corresponding `WP_Block_Template` object as context argument for the visitor.
-	// To that end, we need to suppress hooked blocks from getting inserted into the template.
-	add_filter( 'hooked_block_types', '__return_empty_array', 99999, 0 );
-	$template = $request['id'] ? get_block_template( $request['id'], $post_type ) : null;
-	remove_filter( 'hooked_block_types', '__return_empty_array', 99999 );
+	if ( ! empty( $changes->ID ) ) {
+		$post = get_post( $changes->ID );
+		$terms = get_the_terms( $post, 'wp_theme' );
```

Comment:
[`get_the_terms`](https://developer.wordpress.org/reference/functions/get_the_terms/) accepts a `WP_Post` object, or a post ID:

```suggestion
		$terms = get_the_terms( $changes->ID, 'wp_theme' );
```

Reply:
Ah, oops. We need `$post` further down. I'll revert.